windows/src/main/csharp/ch/cyberduck/ui/ViewModels/TransferViewModel.cs

```diff
@@ -0,0 +1,244 @@
+        [ObservableProperty]
+        private bool progressPending;
+        [ObservableProperty]
+        [NotifyPropertyChangedFor(nameof(ItemStatus))]
+        [NotifyPropertyChangedFor(nameof(MessageText))]
+        [NotifyPropertyChangedFor(nameof(ProgressText))]
+        [NotifyPropertyChangedFor(nameof(Summary))]
+        private TransferProgressModel progressState;
+        [ObservableProperty]
+        [NotifyPropertyChangedFor(nameof(ItemStatus))]
+        [NotifyPropertyChangedFor(nameof(Summary))]
+        private bool running;
+        [ObservableProperty]
+        private TransferItemViewModel selectedTransferItem;
+        [ObservableProperty]
+        private int selectedTransferItemIndex;
+
+        public bool Completed => Transfer.Completed;
+
+        public string ItemStatus => Running ? ProgressText : StatusText;
+
+        public string Local => Transfer.Local;
+
+        public string MessageText => ProgressState switch
+        {
+            { } state => state.Message,
+            _ => Transfer.Timestamp switch
+            {
+                { } date => dateFormatter.getLongFormat(date.getTime(), false),
+                _ => null,
+            },
+        };
+
+        public string Name => Transfer.Name;
+
+        public string ProgressText => ProgressState switch
+        {
+            { } state => state.Text,
+            _ => string.Format(
+                locale.localize("{0} of {1}", "Localizable"),
+                sizeFormatter.format(Transfer.Transferred),
+                sizeFormatter.format(Transfer.Size)),
+        };
+
+        public string StatusText => Transfer.Completed
+            ? locale.localize(string.Format("{0} complete", transferTypeName), "Status")
+            : locale.localize("Transfer incomplete", "Status");
+
+        public string Summary
+        {
+            get
+            {
+                StringBuilder builder = new();
+
+                if (Running)
+                {
+                    builder.AppendLine(locale.localize("Transfer in progress", "Localizable"));
+                }
+                else
+                {
+                    builder.AppendLine(StatusText);
+                    builder.AppendLine(ProgressText);
+                }
+
+                return builder.ToString();
+            }
+        }
+
+        public TransferModel Transfer { get; }
+
+        public TransferDirection TransferDirection { get; }
+
+        public ReadOnlyObservableCollection<TransferItemViewModel> Roots => roots;
+
+        public TransferViewModel(TransferController controller, TransferModel transfer, Locale locale)
+        {
+            this.locale = locale;
+            Transfer = transfer;
+
+            transferTypeName = CultureInfo.CurrentCulture.TextInfo.ToTitleCase(Transfer.Model.getType().name());
+
+            var progressState = Observable.Create<TransferProgressModel>(observer =>
+            {
+                bool emitEmpty = true;
```

Comment:
Cannot work around. The Controller.Progress is an ObservableCache, which allows removal/adding of new entries.